include/envoy/network/connection.h

```diff
@@ -254,6 +254,21 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    */
   virtual uint32_t bufferLimit() const PURE;
 
+  /**
+   * Set the receive buffer size for the socket to a value specified by the 'buffer_size'
```

Comment:
Please split this out into a separate change with dedicated tests.

Reply:
I will remove these changes from this patch-set and raise a separate PR. 

If the interpretation of my previous comment is inline with your thinking of design, this is not required in this commit.

Thanks!
Rama